applications/nrf5340_audio/src/bluetooth/le_audio_cis_gateway.c

```diff
@@ -183,7 +216,14 @@ static void add_remote_codec(struct bt_codec *codec, int index, uint8_t type)
 static void discover_sink_cb(struct bt_conn *conn, struct bt_codec *codec, struct bt_audio_ep *ep,
 			     struct bt_audio_discover_params *params)
 {
-	int ret;
+	int ret = 0;
+	int ep_index = 0;
+
+	if (conn == headset_conn[AUDIO_CHANNEL_LEFT]) {
+		ep_index = AUDIO_CHANNEL_LEFT;
+	} else if (conn == headset_conn[AUDIO_CHANNEL_RIGHT]) {
+		ep_index = AUDIO_CHANNEL_RIGHT;
+	}
```

Comment:
What happens if none of these two if conditions are true? Then ep_index is 0.

Is AUDIO_CHANNEL_LEFT = 0? Then you should probably declare the variable like this:
int ep_index = AUDIO_CHANNEL_LEFT, and only check for AUDIO_CHANNEL_RIGHT (only one if and no else if)

Reply:
Will add the check for unknown connection (neither left channel nor right channel)